instrumentation/netty-reactor-0.9.0/src/main/java/reactor/core/publisher/LambdaMonoSubscriber_Instrumentation.java

```diff
@@ -0,0 +1,45 @@
+package reactor.core.publisher;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Token;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.WeaveAllConstructors;
+import com.newrelic.api.agent.weaver.Weaver;
+import reactor.util.context.Context;
+
+@Weave(originalName = "reactor.core.publisher.LambdaMonoSubscriber")
+abstract class LambdaMonoSubscriber_Instrumentation {
+    @NewField
+    private Context nrContext;
+    final Context initialContext = Weaver.callOriginal();
+
+    @WeaveAllConstructors
+    protected LambdaMonoSubscriber_Instrumentation() {
+        // LamdaMonoSubscriber creates a new Context, so we create a new token and put it on the Context
+        // to be linked by TokenLinkingSubscriber but expired on onComplete here
+        if (AgentBridge.getAgent().getTransaction(false) != null
+                && initialContext.getOrDefault("newrelic-token", null) == null) {
+            nrContext = Context.of("newrelic-token", NewRelic.getAgent().getTransaction().getToken());
```

Comment:
I believe we discussed naming this `newrelic-token` field to something else to disambiguate it from the key of the other context.

Reply:
Doing so made the code considerably more complicated and brittle. This way we continue to use the TokenLinkingSubscriber rather a new Subscriber.